platform/x11/cog-platform-x11.c

```diff
@@ -0,0 +1,896 @@
+
+#include <cog.h>
+
+#include <EGL/egl.h>
+#include <EGL/eglext.h>
+#include <GLES2/gl2.h>
+#include <GLES2/gl2ext.h>
+
+#include <assert.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <wpe/webkit.h>
+#include <wpe/fdo.h>
+#include <wpe/fdo-egl.h>
+
+#include <xcb/xcb.h>
+#include <xkbcommon/xkbcommon-x11.h>
+#include <X11/Xlib.h>
+#include <X11/Xlib-xcb.h>
+
+
+#ifndef EGL_EXT_platform_base
+#define EGL_EXT_platform_base 1
+typedef EGLDisplay (EGLAPIENTRYP PFNEGLGETPLATFORMDISPLAYEXTPROC) (EGLenum platform, void *native_display, const EGLint *attrib_list);
+typedef EGLSurface (EGLAPIENTRYP PFNEGLCREATEPLATFORMWINDOWSURFACEEXTPROC) (EGLDisplay dpy, EGLConfig config, void *native_window, const EGLint *attrib_list);
+typedef EGLSurface (EGLAPIENTRYP PFNEGLCREATEPLATFORMPIXMAPSURFACEEXTPROC) (EGLDisplay dpy, EGLConfig config, void *native_pixmap, const EGLint *attrib_list);
+#endif
+
+#ifndef EGL_EXT_platform_x11
+#define EGL_EXT_platform_x11 1
+#define EGL_PLATFORM_X11_EXT              0x31D5
+#define EGL_PLATFORM_X11_SCREEN_EXT       0x31D6
+#endif /* EGL_EXT_platform_x11 */
+
+#ifndef GL_OES_EGL_image
+#define GL_OES_EGL_image 1
+typedef void *GLeglImageOES;
+typedef void (GL_APIENTRYP PFNGLEGLIMAGETARGETTEXTURE2DOESPROC) (GLenum target, GLeglImageOES image);
+typedef void (GL_APIENTRYP PFNGLEGLIMAGETARGETRENDERBUFFERSTORAGEOESPROC) (GLenum target, GLeglImageOES image);
+#endif
+
+#define DEFAULT_WIDTH  1024
+#define DEFAULT_HEIGHT  768
+
+#if defined(WPE_CHECK_VERSION)
+# define HAVE_2D_AXIS_EVENT WPE_CHECK_VERSION(1, 5, 0) && WEBKIT_CHECK_VERSION(2, 27, 4)
+#else
+# define HAVE_2D_AXIS_EVENT 0
+#endif /* WPE_CHECK_VERSION */
```

Comment:
Fine with that, but it should be done separately since this 2D-axis-event check alone is done in multiple modules and in the core application/library.

Reply:
Sure, let's do it as a follow-up later on, I've filed issue #241 to track this one :+1: